### ISSUE TITLE
Remove `MerkleRoot` interface

### ISSUE THREAD

**Issue by thenswan** [hashgraph/hedera-services#16436]
After #15638 there will be no need to keep `MerkleRoot` interface. This issue also depends on #16322. The execution plan:
   - `SwirldState getSwirldState()` can be superseded by lifecycle methods in `MerkleStateRoot`;
   - `getReadablePlatformState`, `getWritablePlatformState` and `updatePlatformState` will be redundant after #16352, so it should be easy to simplify things;
   - `getInfoString` should be moved to the `MerkleInternal`;
   -  `State` interface also has `copy` and `createSnapshot`, so they can be simply deleted.
